src/clue.js

```diff
@@ -2,25 +2,191 @@
+        color: "white",
+    },
+    {
+        firstName: "Victor",
+        lastName: "Plum",
+        occupation: "Designer",
+        age: 22,
+        description: "Billionaire video game designer",
+        image: "https://66.media.tumblr.com/ee7155882178f73b3781603f0908617c/tumblr_phhxc7EhPJ1w5fh03_540.jpg",
+        color: "purple",
+    },
+    {
+        firstName: "Kasandra",
+        lastName: "Scarlet",
+        occupation: "Actor",
+        age: 31,
+        description: "She is an A-list movie star with a dark past",
+        image: "https://www.radiotimes.com/uploads/images/Original/111967.jpg",
+        color: "red",
+    },
+    {
+        firstName: "Eleanor",
+        lastName: "Peacock",
+        occupation: "Socialité",
+        age: 36,
+        description: "She is from a wealthy family and uses her status and money to earn popularity",
+        image: "https://metrouk2.files.wordpress.com/2016/07/mrs-peacock.jpg",
+        color: "blue",
+    },
+    {
+        firstName: "Jack",
+        lastName: "Mustard",
+        occupation: "Retired Football player",
+        age: 62,
+        description: "He is a former football player who tries to get by on his former glory",
+        image: "https://static.independent.co.uk/s3fs-public/thumbnails/image/2016/07/04/08/unspecified-3.jpg",
+        color: "yellow",
+    }
+  ];
 
-// Rooms Array
+  // Weapons Array
 
-const roomsArray = [];
+const weaponsArray = [
+    {
+        name: "rope",
+        weight: 10,
+    },
+    {
+        name: "knife",
+        weight: 8,
+    },
+    {
+        name: "candlestick",
+        weight: 2,
+    },
+    {
+        name: "dumbbell",
+        weight: 30,
+    },
+    {
+        name: "poison",
+        weight: 2,
+    },
+    {
+        name:"axe", 
+        weight: 15,
+    },
+    {
+        name: "bat",
+        weight: 13,
+    },
+    {
+        name: "trophy",
+        weight: 25,
+    },
+    {
+        name: "pistol",
+        weight: 20,
+    }
+];
 
-// Weapons Array
+// Rooms Array
 
-const weaponsArray = [];
+const roomsArray = [
+    {
+        name: "Dining Room",
+    },
+    {
+        name: "Conservatory",
+    },
+    {
+        name: "Kitchen",
+    },
+    {    
+        name: "Study",
+    },
+    {
+        name: "Library",
+    },
+    {
+        name: "Billiard Room",
+    },
+    {
+        name: "Lounge",
+    },
+    {
+        name: "Ballroom",
+    },
+    {
+        name: "Hall",
+    },
+    {
+        name: "Spa",
+    },
+    {
+        name: "Living Room",
+    },
+    {
+        name: "Observatory",
+    },
+    {
+        name: "Theater",
+    },
+    {
+        name: "Guest House",
+    },
+    {
+        name: "Patio",
+    }
+];
 
 
 // ITERATION 2
 
-function selectRandom() {}
+function selectRandom(arr) {
+    if (arr === 0) {
+        return undefined;
+    }
+     else if (arr.length === 0) {
+        return arr[0];
+     }
+     const randomElement = Math.floor(Math.random() * arr.length);
```

Comment:
This is correct but actually there is no need for the conditions here, you can just do:

```
function selectRandom(arr) {
           const randomIndex = Math.floor(Math.random() * arr.length);
           return arr[randomIndex];
}
```
Because if the array is empty randomIndex will be equal to 0 and since there is no index 0 `[0]` it will already return `undefined`. And if we only have 1 element inside the array `randomIndex` will be 0 again but this time we have the index 0 `[0]` so we will return it.